agent/consul/catalog_endpoint.go

```diff
@@ -284,13 +286,17 @@ func (c *Catalog) ListNodes(args *structs.DCSpecificRequest, reply *structs.Inde
 		})
 }
 
+func isUnmodified(opts structs.QueryOptions, index uint64) bool {
+	return opts.AllowNotModifiedResponse && opts.MinQueryIndex > 0 && opts.MinQueryIndex == index
```

Comment:
In #7995 this was `opts.MinQueryIndex > 1`, but my test was failing. `opts.MinQueryIndex > 0` should be correct, right?

Reply:
It seems correct to me. MinQueryIndex of 0 would indicate its not a blocking query.

Reply:
Well actually, at startup, many things do happen, and there are several parts where there are weird index in tests... Might be an issue, but not sure